src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -130,22 +126,36 @@ public ElasticsearchClient(
                     .toArray(new HttpHost[config.connectionUrls().size()])
             )
             .setHttpClientConfigCallback(configCallbackHandler)
-            .setRequestConfigCallback(configCallbackHandler)
     );
     this.bulkProcessor = BulkProcessor
-        .builder((req, lis) -> client.bulkAsync(req, RequestOptions.DEFAULT, lis), buildListener())
+        .builder(buildConsumer(), buildListener())
         .setBulkActions(config.batchSize())
         .setConcurrentRequests(config.maxInFlightRequests() - 1) // 0 = no concurrent requests
         .setFlushInterval(TimeValue.timeValueMillis(config.lingerMs()))
-        .setBackoffPolicy(
-            BackoffPolicy.exponentialBackoff(
-                TimeValue.timeValueMillis(config.retryBackoffMs()),
-                config.maxRetries()
-            )
-        )
+        // Disabling bulk processor retries, because they only cover a small subset of errors
+        // (see https://github.com/elastic/elasticsearch/issues/71159)
+        // We are doing retries in the async thread instead.
+        .setBackoffPolicy(BackoffPolicy.noBackoff())
         .build();
   }
 
+  private BiConsumer<BulkRequest, ActionListener<BulkResponse>> buildConsumer() {
+    return (req, lis) ->
+      bulkExecutorService.submit(() -> {
```

Comment:
The result of this submit call is not used, and this merely enqueues the operation and returns. Should this method use `bulkExecutorService.submit(...).get();` to block until the request is actually processed? If not, what is providing the backpressure to the Connect call to `SinkTask.put(...)` if the bulk processor doesn't fill up?

Reply:
As discussed, concurrency is limited by BulkProcessor, and the result is reported via the ActionLister. Added documentation to explain all this.